frappe/public/js/frappe/form/controls/link.js

```diff
@@ -571,6 +577,38 @@ frappe.ui.form.ControlLink = class ControlLink extends frappe.ui.form.ControlDat
 			$.extend(args.filters, this.df.filters);
 		}
 	}
+
+	apply_link_field_filters() {
+		let link_filters = JSON.parse(this.df.link_filters);
+		let filters = this.parse_filters(link_filters);
+		// take filters from the link field and add to the query
+		this.get_query = function () {
+			return {
+				filters,
+			};
+		};
+	}
+
+	parse_filters(link_filters) {
+		let filters = {};
+		link_filters.forEach((filter) => {
+			let [_, fieldname, operator, value] = filter;
+			value = String(value).replace(/%/g, "");
+			if (value.startsWith("eval:")) {
+				// get the value to calculate
+				value = value.split("eval:")[1];
```

Comment:
```suggestion
			if (value.startsWith("eval:")) {
				value = String(value).replace(/%/g, "");
				// get the value to calculate
				value = value.split("eval:")[1];
```

Reply:
Can't just simply swap the lines, 
the regex line will have to be there only, 

reason:
when the operator is "like" the value is stored as "%eval:doc.fieldname%", so the 
`value.startsWith("eval:")` will result in false, hence will never get inside that if block.

maybe I can check if the operator is "like"
then only apply the regex line

what do you think @cogk @ankush ?

Reply:
Ah I understand.

```suggestion
			if (String(value).matches("%?eval:")) {
				value = String(value).replace(/%/g, "");
				// get the value to calculate
				value = value.split("eval:")[1];
```

Maybe we could do some additional cleanup directly in the Form Builder too?

Reply:
we can just check if the operator's value is "like", then do the cleanup